diff --git "a/speedtest.widget.php" "b/speedtest.widget.php"
index e12f9a0..f35fe49 100644
--- "a/speedtest.widget.php"
+++ "b/speedtest.widget.php"
@@ -65,8 +65,8 @@
 // Replaces French character HTML entities
 function fix_french_entities($results) {
 
-	$accents = ['/À/','/à/','/Â/','/â/','/Ç/','/ç/','/É/','/é/','/È/','/è/','/Ê/','/ê/','/Ë/','/ë/','/Î/','/î/','/Ï/','/ï/','/Ô/','/ô/'];
-	$replace = ['A','a','A','a','C','c','E','e','E','e','E','e','E','e','I','i','I','i','O','o'];
+	$accents = ['/À/','/à/','/Â/','/â/','/Ç/','/ç/','/É/','/é/','/È/','/è/','/Ê/','/ê/','/Ë/','/ë/','/Î/','/î/','/Ï/','/ï/','/Ô/','/ô/','/Ù/','/ù/','/Û/','/û/'];
+	$replace = ['A','a','A','a','C','c','E','e','E','e','E','e','E','e','I','i','I','i','O','o','U','u','U','u'];
 
 	return preg_replace($accents, $replace, $results);
 }
